tests/Unit/NumericalAlgorithms/Spectral/Python/Test_Mesh.py

```diff
@@ -91,6 +96,10 @@ def test_nonuniform_all(self):
             m2 = Mesh2D(extents2D, bases2D, quadratures2D)
             m3 = Mesh3D(extents3D, bases3D, quadratures3D)
 
+            m1 = pickle.loads(pickle.dumps(m1, 2))
+            m2 = pickle.loads(pickle.dumps(m2, 2))
+            m3 = pickle.loads(pickle.dumps(m3, 2))
```

Comment:
delete the `2`s (just use the default protocol)

Reply:
I have a segfault with python2 after the fixup, before the fixup this worked. I assume this must be related to removing the protocol here?

Reply:
Found this in the docs:

> The second argument to dumps is also crucial: it selects the pickle protocol version 2, since the older version 1 is not supported. Newer versions are also fine—for instance, specify -1 to always use the latest available version. Beware: failure to follow these instructions will cause important pybind11 memory allocation routines to be skipped during unpickling, which will likely lead to memory corruption and/or segmentation faults.
> https://pybind11.readthedocs.io/en/stable/advanced/classes.html#pickling-support

I suggest you write `protocol=2` and a brief comment above that says something like "Use pickle protocol 2 for Py2 compatibility"